### PR TITLE
Trademark Search API 

### PR DESCRIPTION
**Pull request by HanilJain**
### New Trademark Search API

Due to Multiple Merge conflicts and linting issue I have to create this PR to replace #1818 

- it consistent all the suggestion and reviews and it's changes 
- no merge conflicts 
- no pre-commit issues 
- features similar to the previous PR

### CODE REVIEW

File: website/views.py [hunk 1]
```diff
@@ -3711,6 +3711,40 @@ def invite_friend(request):
...
+        slug = request.POST.get("query")
+        return redirect("trademark_detailview", slug=slug)
+    return render(request, "trademark_search.html")
+
+
+def trademark_detailview(request, slug):
+    if settings.USPTO_API is None:
+        return HttpResponse("API KEY NOT SETUP")
+    
+    trademarkAvailable_url = "https://uspto-trademark.p.rapidapi.com/v1/trademarkAvailable/%s" % (
```
Comment (arkid15r) [inline website/views.py:3726]: Here and below: this must be snake-cased.

**Comment by HanilJain** [comment]
@DonnieBLT @arkid15r @AtmegaBuzz 

I have closed the PR #1818 due to complicated merge conflicts issues 
and integrated all the changes you have suggested in this PR.
you can please review.

**Also regarding the API KEY**
I'll create one and share you personally on slack.
but there is limit to free API on it 500 / month and 1000 requests per hour.

File: website/views.py [hunk 2]
```diff
@@ -3711,6 +3711,40 @@ def invite_friend(request):
     }
     return render(request, "invite_friend.html", context)
 
+def trademark_search(request, **kwargs):
+    print(request.method)
```
Comment (arkid15r) [inline website/views.py:3715]: -print

File: website/views.py (hunk 1 above)
Comment (HanilJain) [reply website/views.py:3726]: you mean Trademark_Available_Url and Trademark_Available_Response

File: website/views.py (hunk 2 above)
Comment (HanilJain) [reply website/views.py]: oops!!  Done 👍

File: website/views.py (hunk 1 above)
Comment (arkid15r) [reply website/views.py]: `trademark_available_url`

Comment (HanilJain) [reply website/views.py]: Done 👍

Comment (arkid15r) [reply website/views.py]: I don't think it's done -- you have a mix of camel/snake case currently. https://www.pluralsight.com/blog/software-development/programming-naming-conventions-explained

Comment (HanilJain) [reply website/views.py]: @arkid15r Ohkay got that !
Thankyou could you please see if this was required ?

Comment (arkid15r) [reply website/views.py]: Yes, that was it 👍

**Review by DonnieBLT — APPROVED** [review]

**Pull request merged**